### ISSUE TITLE
radio/checkbox with .input-sm 

### ISSUE THREAD

**Issue by Yahasana** [twbs/bootstrap#18748]
there is 4px margin from top and make the line ship down.
see http://jsbin.com/vixilabewe/edit?html,css,output

**Comment by twbs-lmvtfy** [comment]
Hi @Yahasana!

You appear to have posted a live example (http://jsbin.com/vixilabewe/edit), which is always a good first step. However, according to [the HTML5 validator](http://html5.validator.nu), **your example has some validation errors**, which might potentially be causing your issue:
* line 42, column 5 thru column 10: Stray end tag ```` div ````.

You'll need to **fix these errors** and post a revised example before we can proceed further.
Thanks!

(*Please note that this is a [fully automated](https://github.com/cvrebert/lmvtfy) comment.*)

**Comment by cvrebert** [comment]
After removing that stray `</div>`: http://jsbin.com/jezakel/edit?html,css,output

**Comment by twbs-lmvtfy** [comment]
Hi @cvrebert!

You appear to have posted a live example (http://jsbin.com/jezakel/edit), which is always a good first step. However, according to [Bootlint](https://github.com/twbs/bootlint), **your example has some Bootstrap usage errors**, which might potentially be causing your issue:
* line 12, column 5: [E003](https://github.com/twbs/bootlint/wiki/E003): Found one or more `.row`s that were not children of a grid column or descendants of a `.container` or `.container-fluid`
* line 17, column 5: [E003](https://github.com/twbs/bootlint/wiki/E003): Found one or more `.row`s that were not children of a grid column or descendants of a `.container` or `.container-fluid`
* line 29, column 5: [E003](https://github.com/twbs/bootlint/wiki/E003): Found one or more `.row`s that were not children of a grid column or descendants of a `.container` or `.container-fluid`
* line 37, column 5: [E003](https://github.com/twbs/bootlint/wiki/E003): Found one or more `.row`s that were not children of a grid column or descendants of a `.container` or `.container-fluid`
* line 22, column 9: [E042](https://github.com/twbs/bootlint/wiki/E042): `.form-control` cannot be used on non-textual `<input>`s, such as those whose `type` is: `file`, `checkbox`, `radio`, `range`, `button`
* line 34, column 9: [E042](https://github.com/twbs/bootlint/wiki/E042): `.form-control` cannot be used on non-textual `<input>`s, such as those whose `type` is: `file`, `checkbox`, `radio`, `range`, `button`
* [W002](https://github.com/twbs/bootlint/wiki/W002): `<head>` is missing X-UA-Compatible `<meta>` tag that disables old IE compatibility modes
* [W003](https://github.com/twbs/bootlint/wiki/W003): `<head>` is missing viewport `<meta>` tag that enables responsiveness

You'll need to **fix these errors** and post a revised example before we can proceed further.
Thanks!

(*Please note that this is a [fully automated](https://github.com/cvrebert/lmvtfy) comment.*)

**Comment by cvrebert** [comment]
As noted by the bot, your usage of `.form-control` is invalid.

**Comment by Yahasana** [comment]
is there any markup solution to make radio/checkbox `{sm, lg}`?

**Comment by cvrebert** [comment]
Not that I know of offhand. I've filed a feature request for it for Bootstrap v4 though: #18761

**Comment by cvrebert** [comment]
Depending on your use case, input groups might work; see http://getbootstrap.com/components/#input-groups-checkboxes-radios

**Comment by Yahasana** [comment]
@cvrebert if you double check the checkbox/radio in the input group addon, you will find that they are not vertical align in middle at all

PS: Firefox 43.0.3 in WinXP sp3

**Comment by cvrebert** [comment]
Eh, it's not perfect, but it's not terrible either.
Anyway, checkbox/radio alignment is WontFix for v3 since it's such a can of worms. See https://github.com/twbs/bootstrap/issues/13936#issuecomment-48250465

**Comment by Yahasana** [comment]
thanks for yours patient explanation. I fixed it by myself. whoa!
